client/js/components/snippets/snippet.jsx

```diff
@@ -124,22 +130,25 @@ export default class Snippet extends React.Component {
       mime: this.props.language
     };
     let { style } = this.props;
-    let author = this.props.user.name;
-
+    let currentUser = this.getCurrentUser();
+    let author = this.props.user || currentUser;
+    let authorName = (author && author.name) || 'author';
```

Comment:
I agree with @vrael560. It's definitly a place to `if fix` session problems. It could be hiding some bigger problem.